x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportOpenJobAction.java

```diff
@@ -158,6 +158,10 @@ static void validate(String jobId, Job job) {
                                                                             int maxMachineMemoryPercent,
                                                                             MlMemoryTracker memoryTracker,
                                                                             Logger logger) {
+        if (job == null) {
+            logger.debug("[{}] select node job is null", jobId);
+        }
```

Comment:
maybe assert instead?

Reply:
`job == null` is valid for the rolling upgrade as that field isn't present in the persistent task parameters for jobs < v6.6.0. I added this to help me understand what was happening during the rolling upgrade tests it probably shouldn't be in the released code however.